### PR TITLE
VIM-4425: Removed all iOS7 logic from the library.

### PR DESCRIPTION
**Pull request by alfiehanssen**
#### Ticket
[VIM-4425](https://vimean.atlassian.net/browse/VIM-4425)

#### Ticket Summary
Remove all iOS7-specific logic and classes from VimeoUpload.

#### Implementation Summary
I did the above work. This meant removing all checks for iOS8 and all references to ALAsset APIs.

#### How to Test
Test as part of the [companion PR](https://github.vimeows.com/MobileApps/Vimeo-iOS/pull/968).

### CODE REVIEW

File: VimeoUpload/Descriptor System/Descriptor.swift [hunk 1]
```diff
@@ -78,37 +78,10 @@ class Descriptor: NSObject, NSCoding
...
         {
-            if let identifier = self.currentTaskIdentifier,
-                let task = sessionManager.taskForIdentifier(identifier)
-            {
-                task.resume()
-            }
-        }
-        else
-        {
-            // See note above taskForIdentifierWorkaround for details on why this is necessary (iOS7 bug) [AH] 2/5/2016
```
Comment (huebnerob) [inline VimeoUpload/Descriptor System/Descriptor.swift]: 🐞 💥

File: VimeoUpload/Networking/VimeoSessionManager.swift [hunk 2]
```diff
@@ -83,26 +83,4 @@ class VimeoSessionManager: AFHTTPSessionManager
     {
         fatalError("init(coder:) has not been implemented")
     }
-        
-    // MARK: Private API
-
```
Comment (huebnerob) [inline VimeoUpload/Networking/VimeoSessionManager.swift]: this will also simplify some VimeoNetworking logic IIRC

**Comment by huebnerob** [comment]
DWR @alfiehanssen, dope 👍

**Comment by alfiehanssen** [comment]
Thanks @huebnerob!

**Pull request merged**